src/internal.c

```diff
@@ -5876,7 +5876,10 @@ int SetSSL_CTX(WOLFSSL* ssl, WOLFSSL_CTX* ctx, int writeDup)
     ret = WOLFSSL_SUCCESS; /* set default ret */
 
     ssl->ctx     = ctx; /* only for passing to calls, options could change */
-    ssl->version = ctx->method->version;
+    /* Don't change version on a SSL object that has already started a
+     * handshake */
+    if (!ssl->msgsReceived.got_client_hello)
```

Comment:
What prompted adding this? What about client side? It doesn't mark `msgsReceived.got_client_hello`?

Reply:
Apache is changing the SSL context in the SNI callback and resetting the version breaks downgrading. I'll add the server side equivalent.